pkg/webhook/controlplane/ensurer.go

```diff
@@ -125,27 +127,35 @@ func (e *ensurer) EnsureKubeletServiceUnitOptions(ctx context.Context, gctx gcon
 	return new, nil
 }
 
-func ensureKubeletCommandLineArgs(command []string) []string {
+func ensureKubeletCommandLineArgs(command []string, kubeletVersion *semver.Version) []string {
 	// TODO: Figure out how to provide the provider-id via the kubelet config file (as of Kubernetes 1.19 the kubelet config
 	// offers a new `providerID` field which can be used, and it's expected that `--provider-id` will be deprecated eventually).
 	// Today, the problem is that the provider ID is determined dynamically using the script above, but the kubelet config cannot
 	// reference environment variables like it's possible today with the CLI parameters.
 	// See https://github.com/kubernetes/kubernetes/pull/90494
 	command = extensionswebhook.EnsureStringWithPrefix(command, "--provider-id=", "${PROVIDER_ID}")
-	command = extensionswebhook.EnsureStringWithPrefix(command, "--cloud-provider=", "external")
-	command = extensionswebhook.EnsureStringWithPrefix(command, "--enable-controller-attach-detach=", "true")
+
+	if !version.ConstraintK8sGreaterEqual123.Check(kubeletVersion) {
+		command = extensionswebhook.EnsureStringWithPrefix(command, "--cloud-provider=", "external")
```

Comment:
Hi @rfranzke ,
is "--cloud-provider" argument not needed at all? 
It is still kept in aws extension even if k8s version is older than 1.23
https://github.com/gardener/gardener-extension-provider-aws/blob/master/pkg/webhook/controlplane/ensurer.go#L425

Reply:
> is "--cloud-provider" argument not needed at all?

In my tests, I didn't see any issue when not providing it, and since it's deprecated with 1.23 I decided to longer provide it. Do you see any issues?

> It is still kept in aws extension even if k8s version is older than 1.23
https://github.com/gardener/gardener-extension-provider-aws/blob/master/pkg/webhook/controlplane/ensurer.go#L425

No, actually only when CSI is disabled which is only the case for < 1.18 AWS clusters, or when CSI is enabled but version is < 1.23

Reply:
ok.  Then it's safe to remove it